src/server.c

```diff
@@ -6158,13 +6179,15 @@ sds genValkeyInfoString(dict *section_dict, int all_sections, int everything) {
         if (sections++) info = sdscat(info, "\r\n");
         info = sdscatprintf(info, "# Keyspace\r\n");
         for (j = 0; j < server.dbnum; j++) {
+            serverDb *db = server.db[j];
+            if (db == NULL) continue;
```

Comment:
This breaks the existing behavior. Now, databases that were never used or initialized will not be logged. We should either pretend the DB exists and print hardcoded zero values for NULL databases, or choose to log only non-empty databases, and document this in the release notes.